### PR TITLE
feat: add 4626 core code

### PR DESCRIPTION
**Pull request by Ninjagod1251**
I add the ERC4626 methods.

We can pre and post generate ERC4626.json with these links for pre and post hooks:
https://cookiecutter.readthedocs.io/en/stable/advanced/hooks.html#example-conditional-files-directories
https://cookiecutter.readthedocs.io/en/stable/advanced/hooks.html

### CODE REVIEW

**Comment by Ninjagod1251** [comment]
Please let me know if it is too simple.

Next will be the test

**Comment by Ninjagod1251** [comment]
Mint is different in ERC20 Default vs ERC4626. Let me know how I should implement the change. 

The way implemented it would be the easiest way.

File: {{cookiecutter.project_name}}/contracts/Token.vy [hunk 1]
```diff
@@ -123,11 +241,28 @@ def burn(amount: uint256):
     self.balanceOf[msg.sender] -= amount
     self.totalSupply -= amount
 
-    log Transfer(msg.sender, ZERO_ADDRESS, amount)
+    log Transfer(msg.sender, empty(address), amount)
 {%- endif %}
 {%- if cookiecutter.mintable == "y" %}
 
+{%- if cookiecutter.ERC4626 == 'y' %}
```
Comment (fubuloubu) [inline {{cookiecutter.project_name}}/contracts/Token.vy]: I think these options might need to be mutually exclusive

**Comment by fubuloubu** [comment]
Tests need updating to include `"ERC4626"` template field: https://github.com/ApeAcademy/ERC20/blob/main/tests/Basic.json

File: {{cookiecutter.project_name}}/contracts/Token.vy [hunk 2]
```diff
@@ -1,8 +1,15 @@
-# @version 0.3.3
+# @version 0.3.4
```
Comment (fubuloubu) [inline {{cookiecutter.project_name}}/contracts/Token.vy]: 0.3.7 came out yesterday

**Review by fubuloubu — DISMISSED** [review]

File: .github/workflows/test.yaml [hunk 3]
```diff
@@ -23,11 +24,15 @@ jobs:
...
         ape plugins install -U .
+    - name: install additional dependencies
+      if: matrix . template == "ERC4626"
+      run: npm install --save-dev hardhat
     - name: Compile contracts
       run: |
         cd ${{ matrix.template }}
         ape compile --size
     - name: Run tests
+      with:
```
Comment (fubuloubu) [inline .github/workflows/test.yaml]:
```suggestion
      env:
        ETHERSCAN_API_KEY: ${{ secrets.ETHERSCAN_API_KEY }}
        WEB3_ALCHEMY_PROJECT_ID: ${{ secrets.WEB3_ALCHEMY_PROJECT_ID }}
```

File: .github/workflows/test.yaml [hunk 4]
```diff
@@ -23,11 +24,15 @@ jobs:
       run: |
         cd ${{ matrix.template }}
         ape plugins install -U .
+    - name: install additional dependencies
+      if: matrix . template == "ERC4626"
```
Comment (fubuloubu) [inline .github/workflows/test.yaml]:
```suggestion
      if: matrix.template == "ERC4626"
```

**Review by fubuloubu — CHANGES_REQUESTED** [review]
You included the temp folder by mistake?

File: .github/workflows/test.yaml [hunk 5]
```diff
@@ -23,11 +24,17 @@ jobs:
       run: |
         cd ${{ matrix.template }}
         ape plugins install -U .
+    - name: install additional dependencies
+      if: matrix.template == "ERC4626"
```
Comment (fubuloubu) [inline .github/workflows/test.yaml]:
```suggestion
      if: ${{ matrix.template == 'ERC4626' }}
```

**Review by fubuloubu — APPROVED** [review]

**Pull request merged**